src/Sylius/Bundle/CoreBundle/Resources/config/services/providers.xml

```diff
@@ -89,5 +89,11 @@
             <argument type="service" id="sylius.repository.order" />
             <tag name="sylius.statistics.orders_count_provider" type="year" />
         </service>
+      
+        <service id="sylius.provider.configurable_grid_provider" class="Sylius\Bundle\CoreBundle\Grid\Provider\ConfigurableProvider">
+            <argument type="tagged_locator" tag="sylius.grid_provider" index-by="key" />
+            <argument>%sylius_core.grids_configuration%</argument>
+            <tag name="sylius.grid_provider" key="array" priority="100"/>
```

Comment:
Grid providers are called by [ChainProvider](https://github.com/Sylius/Grid/blob/master/Provider/ChainProvider.php) ordered by `priority`, where first successful call returns result.

Hence we don't need to override [`key="array"`](https://github.com/Sylius/SyliusGridBundle/blob/6a771469d9d28b54c79b5e76576acafef67c80f8/src/Bundle/Resources/config/services.xml#L55C1-L55C75) tag of the `ArrayGridProvider` and can just register this new `ConfigurableGridProvider` with earlier priority.

Same result with no override.

```suggestion
            <tag name="sylius.grid_provider" key="configurable" priority="-300"/>
```